### PR TITLE
feat(dropdown-searchbar): add dropdown and search components

### PR DESCRIPTION
**Pull request by ashinzekene**
#### What does this PR do
Add dropdown and searchbar components based on the [styleguide](https://www.figma.com/file/wtojDaoY2PJED946ZMIvsX/OSCA-style-guide%2Fpattern?node-id=81%3A0)
#### Description of Task to be completed
Add dropdown and searchbar components
#### How should this be manually tested

`npm run storybook`
#### What are the relevant issues this PR belongs to
#15 

#### Any background context you want to add
Marking as WIP because I feel `searchbar` and `dropdown` components should have a label

cc: @perriefidelis @BolajiAyodeji 
#### Screenshots
image
image

### CODE REVIEW

**Comment by BolajiAyodeji** [comment]
Let me know when this is ready for a MR  :+1:

**Comment by ashinzekene** [comment]
> Let me know when this is ready for a MR 👍

Actually still waiting for @perriefidelis to clarify the issue I raised  above

**Comment by BolajiAyodeji** [comment]
> > Let me know when this is ready for a MR +1
> 
> Actually still waiting for @perriefidelis to clarify the issue I raised above

Alright, @perriefidelis please see to this.

**Comment by perriefidelis** [comment]
one more little change ... the search icon should be larger

**Comment by BolajiAyodeji** [comment]
@perriefidelis here's the issue: `Marking as WIP because I feel searchbar and dropdown components should have a label`

**Comment by alabobriggs** [comment]
@ashinzekene I don't see why a search bar needs to have a label it will be within components and wrappers such as `Navbar`. It is not a standalone input like that of text or select. Imagine having a label on the searchbar placed on the navbar (Doesn't sound nice). The dropdown acts like an auto suggestion so a label isn't still required.

**Comment by ashinzekene** [comment]
When an item has been selected from a dropdown it is impossible to know what the dropdown is for because the placeholder would no longer show. Having a label outside might help but fails accessibility. For the navbar example it still work if the label is inline. Besides the label does not need to be required, there should just be a style guide for one if needed

File: src/components/searchbar/index.js [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+  useEffect(() => {
+    document.documentElement.addEventListener('pointerdown', close)
+    return () => document.documentElement.removeEventListener('pointerdown', close)
+  })
+
+  return (
+    <div className={classNames(style.searchbar, props.className)} {...props}>
+      <div className={style.searchbarInput} onPointerDown={() => setIsOpen(true)}>
+        <object aria-label="Search Icon" className={style.searchImg} data={searchImg}>search icon</object>
+        <input type="text" value={textVal} inputMode="search" onChange={handleInput} />
```
[the remark this answers is not in the document]
Comment (ashinzekene) [reply src/components/searchbar/index.js]: Using type search adds some native styling to the input depending on the browser.

https://twitter.com/segunolalive/status/1202477462766141440

File: src/components/searchbar/index.js [hunk 2]
```diff
@@ -0,0 +1,103 @@
+import React, { useMemo, useState, useEffect } from 'react';
+import PropTypes from 'prop-types';
+import style from "./style.module.css";
+import classNames from "classnames";
+
```
[the remark this answers is not in the document]
Comment (ashinzekene) [reply src/components/searchbar/index.js]: Done

**Comment by perriefidelis** [comment]
testing

**Pull request merged**